conf/bigearthnet.yaml

```diff
@@ -0,0 +1,18 @@
+trainer:
+  gpus: 1  # single GPU training
+  min_epochs: 10
+  max_epochs: 40
+  benchmark: True
+
+experiment:
+  task: "bigearthnet"
+  module:
+    loss: "bce"
+    classification_model: "resnet18"
+    learning_rate: 1e-3
+    learning_rate_schedule_patience: 6
+    in_channels: 14
+  datamodule:
+    batch_size: 128
+    num_workers: 6
+    bands: "all"
```

Comment:
Most of these two conf files are duplicated, should they be?

Reply:
I think the purpose is that configs in conf/task_defaults have all the required params to run, but conf/ basically inherits params from conf/task_defaults if they are not specified. So you could essentially have almost no params defined in conf/ configs and it would still run.

Reply:
In that case why do we need two separate files at all? We should have a single file with defaults and users can provide additional files or specify on the CLI if they want to override. @calebrob6's explanation didn't really make sense to me.

Reply:
What part didn't make sense? The `task_defaults/` are default parameters for the trainers. The others are scripts that you can use to run the trainers.

We could reorganize to have `conf/task_defaults/` and `conf/examples/`. Alternatively, you could rewrite the task defaults as dataclasses per https://omegaconf.readthedocs.io/en/2.1_branch/structured_config.html and include them in the code somewhere.

Reply:
> The others are scripts that you can use to run the trainers.

But couldn't you just use the files in `conf/task_defaults` for this? At least from this PR they seem to have the same structure and mostly the same values.

Reply:
Its a convenience thing - you could just not have defaults at all and have to specify everything all the time - I usually prefer to draw the line closer to "will run by default" (these will not run by default because you need to specify the dataset directory). Take a look through all the config and how they map to arguments in `train.py` and the tasks/datamodules.

Reply:
I still wish `./train.py --help` told me which variables I can actually set. I usually have to dig through these conf files, and even then I don't always find the settings I need...

Reply:
But you literally haven't used train.py lol -- currently you just don't like the _idea_ that it doesn't tell you which variables can be set. If you want to tear up the current omegaconf way of doing things, replace it with argparse, and make a way where we can still pass arguments as a yaml files then I'm totally with it.

Reply:
Just my 2 cents, but using config files and omegaconf or hydra is better than argparse as the number of args grows you'll end up just making bash scripts to save which args you used for a run which is effectively uglier than using config files. 

We may be able to get the best of both however by using the "add_model_specific_args" method in the tasks which we can add to the argparse in train.py https://pytorch-lightning.readthedocs.io/en/latest/common/hyperparameters.html.

Reply:
I definitely agree that config files are better than no config files, but if we can get both config files and a good way to document which settings can be modified, that would be even better!

Reply:
Yeah I think adding args to the lightning modules would be incredibly easy by adding the method above to the base tasks. This way they show up in train.py --help and we can keep track in the scripts.

Reply:
Going to leave as is for now. Will submit a new PR adding hparam argparse args.